packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,39 @@ export default declare((api, opts) => {
     }
   }
 
+  function doesAnyChildNodeHaveMoreThanOneProperty(node) {
+    if (node.argument) return true;
+    let innerNode;
+
+    // Array destructing case
+    if (node.elements) {
+      if (node.elements.length > 1) return true;
+      // Recurse on the element
+      innerNode = node.elements[0];
+    }
+
+    // Nested object case
+    if (node.properties) {
+      if (node.properties.length > 1) return true;
+      // Recurse value of the value of the first property
+      innerNode = node.properties[0].value;
+      // Short circuit if the value is falsy
+      if (!innerNode) {
+        return false;
+      }
+    }
```

Comment:
At this point we still need to handle the assignment pattern:

```js
const { x: { ...x } = {} } = z();
```

When we recurse to `{ ...x } = {}`, we need to check the `left` (`{ ...x }`) of the assignment pattern.

Reply:
Added this test case as
```js
const { x20: { ...y20 } = { } } = z();
```